ADR/0039-workspace-deprecation.md

```diff
@@ -0,0 +1,98 @@
+
+* We made the decision (at Red Hat) to stop trying to offer a managed service interface for Konflux. The user signup, waitlist, and workspace provisioning interfaces made a lot of sense there.
+
+## Decision
+
+1. We will remove the Workspace abstraction from Konflux and use standard Kubernetes namespaces
+directly for providing a place for users to run their pipelines and store their configurations
+and secrets.
+
+2. We will stop using the term Workspace, and start to use the term Namespace.
+
+3. The Konflux UI will expose a wizard for creating a new namespace. This
+wizard will be available to any user that has permissions to create namespaces.
+
+4. Konflux will provide an optional automation for creating a namespace for a user
+when he/she logins for the first time.
+
+5. Konflux won't be opinionated about the mechanism for initializing/maintaining namespaces with
+supporting resources such as ResourceQuota and LimitRange. Konflux will differ
+this responsibility to other tools which are specialized in this task such as 
+(but not limited to) [Kyverno](https://kyverno.io/policies/best-practices/add-ns-quota/add-ns-quota/) and [ArgoCD](https://github.com/konflux-ci/namespace-generator).
+
+6. Konflux will provide a thin [backend service](https://github.com/konflux-ci/workspace-manager) for listing the namespaces where the user has at least view access
+to the Konflux CRDs. This list will be used by the namespace switcher in the UI.
+This is required since the Kubernetes API doesn't let the user to list a subset
+of namespace. The user gets permissions to list all namespace or none.
+
+7. Konflux will provide ClusterRoles that will grant permissions to the Konflux
+and Tekton CRDs. Those will be (aggregated)[https://github.com/konflux-ci/konflux-ci/issues/440] to the built-in Kubernetes roles (`view`, `edit`, `admin`).
+
+8. Public viewer access will be provided by assigning the `view` role (see above) to the
+`system:authenticated` built-in group that contains all the authenticated users.
+
+9. Same as Kubernetes, Konflux won't have a resource for representing a user. Instead,
+it will use external Identity providers.
+
+9. Konflux won't provide a way for creating a ephemeral namespaces. It will differ this
+task to another tool.
+
+10. The `join the waitlist` button will be removed from the Konflux UI.
+
+11. Konflux won't provide a transparent multi-cluster deployment.
+If required, existing open source projects that handle multi-cluster deployments
+should be explored and Konflux should integrate with them.
+
+## Consequences
+
+1. By removing the Workspace abstraction, and delegating namespace management tasks to other open source tools,
+the Konflux developers will have more attention for developing features related to to Konflux' core mission.
+
+2. Operating Konflux will become easier for users who already familiar with Kubernetes concepts.
+
+3. Konflux operators can use their preferred way to manage namespaces on their clusters.
```

Comment:
I agree that this will be the same. We can allow multiple ways for the admin/operators to configure the management of namespaces but for the end user it will not matter if they are in kind cluster or a production cluster in AWS the API and user experience is the same

Reply:
@arewm I change the text. Please let me know if it's more clear now.

Reply:
If the are no objections I'll close this thread.